src/modules/mc_att_control/mc_att_control.hpp

```diff
@@ -145,6 +147,10 @@ class MulticopterAttitudeControl : public ModuleBase<MulticopterAttitudeControl>
 	 */
 	matrix::Vector3f pid_attenuations(float tpa_breakpoint, float tpa_rate);
 
+	/** lower yawspeed limit in auto modes because we expect yaw steps */
+	void adapt_auto_yaw_rate_limit();
```

Comment:
That's okay for now, but this changes the closed-loop response.
Let's take the extreme case where the user tightly tuned the yaw P gain to its maximum and is just stable; changing the saturation in auto mode will lead to yaw oscillations or even instability.
I made some modifications in the FlightTask to limit the rate of change of yaw setpoint such that we don't need to do that nasty thing here, I'll try to upstream that really soon.

Reply:
Good point, I totally agree. We definitely need to change that.
- This is refactoring only
- I made the function such that the logic only gets called in the event mode or parameter changes and not in every loop iteration, see review comment https://github.com/PX4/Firmware/pull/11308/files#discussion_r261836721
- I changed it to only switch the yaw rate limit because the rest was just confusing overhead
- The attitude controller currently can produce wobbles in the yaw movement when commanding big steps depending on the tuning gins which currently change the yaw weight (https://github.com/PX4/Firmware/pull/11308/files#diff-c7e26e8b748c35ed8718b27314fa807bR51). This stayed as it is because I'd like to have testing in place first (https://github.com/PX4/Firmware/pull/11573) and then change it according to my findings while experimenting with the prototyping script https://github.com/Auterion/Flight_Control_Prototyping_Scripts/tree/2eef448fb589a1c694e0717b5835644acf8d69dd/quaternion_attitude_control